### PR TITLE
Rework documentation + readme

### PR DESCRIPTION
**Pull request by AlekseySh**

### CODE REVIEW

**Comment by b0nce** [comment]
image

File: README.md [hunk 1]
```diff
@@ -96,25 +65,47 @@ Here are a few examples of such tasks from the computer vision sphere:
...
+Thus, during the training we:
+   1. Use `DataLoader` + `Sampler` to form batches (for example `BalanceSampler`)
+   2. [Only for losses based on combinations] Use `Miner` to form effective pairs or triplets, including those which utilize a memory bank.
+   3. Compute loss.
+
+**Validation part** consists of several steps:
+  1. Accumulating all of the embeddings (`EmbeddingMetrics`).
+  2. Calculating distances between them with respect to query/gallery split.
+  3. Applying some specific retrieval techniques like query reranking or score normalisation.
+  4. Calculating retrieval metrics like *CMC@k*, *Precision@k* or *MeanAveragePrecision*.
```
Comment (DaloroAT) [inline README.md:107]: MeanAveragePrecision`@k`?

**Comment by b0nce** [comment]
[Add link to paper](https://open-metric-learning--153.org.readthedocs.build/en/153/contents/models.html#oml.models.resnet.ResnetExtractor.draw_gradcam), here: [link](https://arxiv.org/abs/1610.02391).

File: tests/test_build_readme.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+from oml.const import PROJECT_ROOT
+
+
+def test_readme_was_built_correctly() -> None:
+    readme_tmp_file = "Readme_tmp"
+    os.chdir(PROJECT_ROOT)
+    os.system(f"make build_readme README_FILE={readme_tmp_file}.md")
+
+    with open(PROJECT_ROOT / "Readme.md", "r") as f1:
```
Comment (DaloroAT) [inline tests/test_build_readme.py:11]: README.md

**Review by DaloroAT — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (AlekseySh) [reply README.md:107]: ok

File: tests/test_build_readme.py (hunk 2 above)
Comment (AlekseySh) [reply tests/test_build_readme.py]: ok

**Pull request merged**